metricbeat/module/windows/perfmon/reader.go

```diff
@@ -137,33 +137,45 @@ func (r *Reader) RefreshCounterPaths() error {
 }
 
 // Read executes a query and returns those values in an event.
-func (r *Reader) Read() ([]mb.Event, error) {
+func (re *Reader) Read(metricset string) ([]mb.Event, error) {
 	// Some counters, such as rate counters, require two counter values in order to compute a displayable value. In this case we must call PdhCollectQueryData twice before calling PdhGetFormattedCounterValue.
 	// For more information, see Collecting Performance Data (https://docs.microsoft.com/en-us/windows/desktop/PerfCtrs/collecting-performance-data).
-	if err := r.query.CollectData(); err != nil {
+	if err := re.query.CollectData(); err != nil {
 		return nil, errors.Wrap(err, "failed querying counter values")
 	}
 
 	// Get the values.
-	values, err := r.query.GetFormattedCounterValues()
+	values, err := re.query.GetFormattedCounterValues()
 	if err != nil {
 		return nil, errors.Wrap(err, "failed formatting counter values")
 	}
+	var events []mb.Event
+	// GroupAllCountersTo config option will only apply to the webserver metricset, where counters for all instances are aggregated
+	if metricsetName != metricset && re.config.GroupAllCountersTo != "" {
```

Comment:
Similarly here, I would prefer to don't depend on the metricset name for certain features, I would prefer this to be more explicit.
Also this generic metricset shouldn't have something specific for the webserver metricset (or any other metricset) if possible.